src/library_dylink.js

```diff
@@ -494,7 +561,7 @@ var LibraryDylink = {
   // If a library was already loaded, it is not loaded a second time. However
   // flags.global and flags.nodelete are handled every time a load request is made.
   // Once a library becomes "global" or "nodelete", it cannot be removed or unloaded.
-  $loadDynamicLibrary__deps: ['$LDSO', '$loadWebAssemblyModule', '$asmjsMangle', '$fetchBinary', '$isInternalSym'],
+  $loadDynamicLibrary__deps: ['$LDSO', '$loadWebAssemblyModule', '$asmjsMangle', '$fetchBinary', '$isInternalSym', '$mergeLibSymbols'],
```

Comment:
Is there a reason to split out `mergeLibSymbols`?   I don't see any other users.   If its just cosmetic lets at least propose that as a separate PR.

Reply:
looking at this again, i should be calling loadDynamicLibrary instead of loadWebAssemblyModule, that way it will merge the symbols properly. 
right now i am doing
```
         var sideExports= loadWebAssemblyModule(compiledModule);
         mergeLibSymbols(sideExports)

```

Reply:
@sbc100 does it make sense to extend loadDynamicLibrary to load a compiled side module so we wont have to expose mergeLibSymbols? the caller to  loadWebAssemblyModule wont be able to use the side module functions without calling mergeLibSymbols anyway.

Reply:
I see. I think perhaps both things are useful.    Its probably worth splitting out that function just for readability, but yes it seems reasonable to to call loadDynamicLibrary from the outside if that is that effect you want.

I wonder if it might be better to use dependency injection (like we do with overriding `instantiateWasm` in worker.js) rather than making the function more polymorphic?   I'm not sure I know that answer but it would be good to be consistent.   Perhaps the same `instantiateWasm` could be used for both side module and main module so you have a single interception point?